app/experimenter/experiments/tests/test_migrations.py

```diff
@@ -1,37 +1,79 @@
 from django_test_migrations.contrib.unittest_case import MigratorTestCase
 
+from experimenter.base.models import Language, Locale
+from experimenter.experiments.models import NimbusExperiment as Experiment
+
 
 class TestMigration(MigratorTestCase):
-    migrate_from = ("experiments", "0210_alter_nimbusexperiment_channel")
-    migrate_to = ("experiments", "0211_alter_nimbusexperiment_targeting_config_slug")
+    migrate_from = ("experiments", "0211_alter_nimbusexperiment_targeting_config_slug")
+    migrate_to = (
+        "experiments",
+        "0212_alter_nimbusexperiment_languages_field_for_mobile_client",
+    )
 
     def prepare(self):
         """Prepare some data before the migration."""
         User = self.old_state.apps.get_model("auth", "User")
+        Locale = self.old_state.apps.get_model("base", "Locale")
+        Language = self.old_state.apps.get_model("base", "Language")
         NimbusExperiment = self.old_state.apps.get_model(
             "experiments", "NimbusExperiment"
         )
         user = User.objects.create(email="test@example.com")
-        NimbusExperiment.objects.create(
+        locale_en = Locale.objects.create(code="en", name="English")
+        Language.objects.create(code="en", name="English")
+        locale_ts = Locale.objects.create(code="ts", name="does not exist")
+
+        desktop_experiment = NimbusExperiment.objects.create(
```

Comment:
Since we only want to run this for draft experiments, we should probably put one in here with a mobile application and some locales and then assert that it doesn't get modified 🙏